Sources/SKCore/CompilationDatabase.swift

```diff
@@ -67,14 +68,28 @@ public protocol CompilationDatabase {
   var allCommands: AnySequence<Command> { get }
 }
 
-/// Loads the compilation database located in `directory`, if any.
+/// Loads the compilation database located in `directory`, if one can be found in `additionalSearchPaths` or in the default search paths of "." and "build".
 public func tryLoadCompilationDatabase(
   directory: AbsolutePath,
+  additionalSearchPaths: [RelativePath] = [],
   _ fileSystem: FileSystem = localFileSystem
 ) -> CompilationDatabase? {
+  let searchPaths =
+    try! additionalSearchPaths + [
+      // These default search paths match the behavior of `clangd`
+      RelativePath(validating: "."),
+      RelativePath(validating: "build"),
+    ]
```

Comment:
Nitpick: I would move `try!` in front of `RelativePath` just to make it clear that the array creation or accessing `additionalPaths` can’t throw. This is just me being overly careful with `try!`.

```suggestion
    additionalSearchPaths + [
      // These default search paths match the behavior of `clangd`
      try! RelativePath(validating: "."),
      try! RelativePath(validating: "build"),
    ]
```